boards/cargo/tock_flags.toml

```diff
@@ -0,0 +1,30 @@
+# Licensed under the Apache License, Version 2.0 or the MIT License.
+# SPDX-License-Identifier: Apache-2.0 OR MIT
+# Copyright Tock Contributors 2024.
+
+[build]
+rustflags = [
+  # Tell rustc to use the LLVM linker. This avoids needing GCC as a dependency
+  # to build the kernel.
+  "-C", "linker=rust-lld",
+  # Use the LLVM lld executable with the `-flavor gnu` flag.
+  "-C", "linker-flavor=ld.lld",
```

Comment:
Should this comment be different? Looks like the "gnu" as flavor came in cd89f10a, but it seems like the comment as-written contradicts the flag?

(Sorry, I know it's not the point of this PR, just more obvious now)

Reply:
I think that comment is correct (https://doc.rust-lang.org/rustc/codegen-options/index.html#linker-flavor):

> ld.lld: use the LLVM lld executable with the [-flavor gnu flag](https://releases.llvm.org/12.0.0/tools/lld/docs/Driver.html) for GNU binutils' ld.